pykafka/simpleconsumer.py

```diff
@@ -646,20 +655,18 @@ def _handle_success(parts):
                         min_bytes=self._fetch_min_bytes
                     )
                 except (IOError, SocketDisconnectedError):
+                    if self._running:
+                        unlock_partitions(iterkeys(partition_reqs))
```

Comment:
I think that if an `owned_partition` was found too full-up in L646 above, it won't end up in `partition_reqs`, and so may remain locked?